gammapy/modeling/models/tests/test_temporal.py

```diff
@@ -319,3 +323,92 @@ def test_plot_constant_model():
     constant_model = ConstantTemporalModel(const=1)
     with mpl_plot_check():
         constant_model.plot(time_range)
+
+
+class MyCustomTemporalModel(TemporalModel):
+    """Temporal model with spectral variability
+
+    F(t) = (E/E0)^-\alpha
+    where \alpha = (\alpha_0 t/t_ref)^-beta
+    """
+
+    tag = "MyCustomTemporalModel"
+    is_energy_dependent = True
+    beta = Parameter("beta", 0.2)
+    _t_ref_default = Time("2000-01-01")
+    t_ref = Parameter("t_ref", _t_ref_default.mjd, unit="day", frozen=True)
+    E0 = Parameter("E0", "1 TeV", frozen=True)
+
+    @staticmethod
+    def evaluate(time, energy, t_ref, beta, E0):
+        alpha = np.power((time / t_ref), -beta)
+        dim = energy / E0
+        if not np.isscalar(dim.value):
+            dim = np.expand_dims(dim, axis=1)
+        return np.power(dim, -alpha)
+
+    def integral(self, t_min, t_max, energy):
+        pars = self.parameters
+        t_ref = Time(pars["t_ref"].quantity, format="mjd")
+        beta = pars["beta"].quantity
+        E0 = pars["E0"].quantity
+        integral = []
+        for t1, t2 in zip(t_min, t_max):
+            integral_ene = []
+            for ene in energy:
```

Comment:
Is the integral method meaningful here?
Should it return the integral over time of the spectral norms or the double integral over time and energy as defined here? 
This does not sound obvious to me.

Reply:
In the case of a normal temporal model, an array of normalisations is returned, ie, one integral for each time interval .

In this case, I am computing the integral between t1 and t2 for each energy. A double integral is not performed.
What do you think @registerrier ?

The implementation is indeed ugly, but this for this I am preparing a PR.